### PR TITLE
refactor: change from pleco to tanton

### PR DESCRIPTION
**Pull request by chase-manning**
# Description

Pleco is no longer maintained, it has outstanding issues and hasn't had a new contribution in two and half years.  
[Tanton](https://github.com/chase-manning/tanton) is a fork of Pleco that is backward compatable and actively maintained.  
This PR changes from using Pleco to using Tanton.

# How Has This Been Tested?

All of the Pleco tests were kept with Tanton and all are passing on `stable`, `alpha` and `nightly` versions of cargo.

The CI build should run in this PR so should test everything needed from the `cdk-rs` side.

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

**Review by sesi200 — COMMENTED** [review]
LGTM, but I'll let @lwshang have the final word

File: examples/chess/src/chess_rs/Cargo.toml
Comment (chase-manning) [inline examples/chess/src/chess_rs/Cargo.toml:19]: Adding support for WebAssembly for getrandom ([Docs](https://docs.rs/getrandom/latest/getrandom/#webassembly-support))

Comment (lwshang) [reply examples/chess/src/chess_rs/Cargo.toml:19]: The canister wasm execution runtime is different from in a browser when it comes to randomness.
There is currently no synchronous randomness API exposed. The only way to get "true" randomness is calling [`raw_rand`](https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-raw_rand) method of the management canister.

Adding `getrandom` here won't make a difference. I'll remove it.

**Review by lwshang — APPROVED** [review]

File: examples/chess/src/chess_rs/Cargo.toml
Comment (chase-manning) [reply examples/chess/src/chess_rs/Cargo.toml]: @lwshang For some reason, without `getrandom = { version = "0.2", features = ["js"] }`, this test fails:

https://github.com/dfinity/cdk-rs/actions/runs/3568798481/jobs/5998064054#step:9:85

Adding it fixes the test though...

Comment (lwshang) [reply examples/chess/src/chess_rs/Cargo.toml]: Yes.

IMO, this is an issue with `tanton` crate. It should have a feature flag to gate its use of `rand`.

I added it back to make compiler happy. While, I believe it will cause problem when using `tanton`'s random related API in canisters.

Comment (lwshang) [reply examples/chess/src/chess_rs/Cargo.toml]: Enable `js` feature of `getrandom` actually adds an import section in the wasm module which is rejected by the IC replica.

So we have to fix `tanton` crate as I said above before we can merge this PR.

Comment (lwshang) [reply examples/chess/src/chess_rs/Cargo.toml]: After more investigation, I believe this is something need to be fixed on `getrandom` side.
It should not forbid compilation without `js` feature enabled.

`pleco` depends on older `rand`/`rand_core` which doesn't introduce `getrandom` indirectly. That's why there is no issue on the main branch.

Comment (lwshang) [reply examples/chess/src/chess_rs/Cargo.toml]: Found that it is possible to register custom `getrandom` function as described [here](https://docs.rs/getrandom/latest/getrandom/macro.register_custom_getrandom.html).

I copied the code and made the example test pass.

**Pull request merged**
